### PR TITLE
feat(modelArts): support notebook service

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
support notebook service

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/modelarts/v1/notebook/requests.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+type CreateOpts struct {
+	Description string         `json:"description,omitempty"`
+	Duration    *int           `json:"duration,omitempty"`
+	Endpoints   []EndpointsReq `json:"endpoints,omitempty"`
+	Feature     string         `json:"feature,omitempty"`
+	Flavor      string         `json:"flavor" required:"true"`
+	ImageId     string         `json:"image_id" required:"true"`
+	Name        string         `json:"name" required:"true"`
+	PoolId      string         `json:"pool_id,omitempty"`
+	Volume      VolumeReq      `json:"volume" required:"true"`
```
Comment (ShiChangkuo) [inline openstack/modelarts/v1/notebook/requests.go:36]: suggest to put **required** args in front of the struct

File: openstack/modelarts/v1/notebook/requests.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	Offset      *int   `q:"offset,omitempty"`
+	Owner       string `q:"owner,omitempty"`
+	SortDir     string `q:"sort_dir,omitempty"`
+	SortKey     string `q:"sort_key,omitempty"`
+	Status      string `q:"status,omitempty"`
+	WorkspaceId string `q:"workspaceId,omitempty"`
+}
+
+type UpdateOpts struct {
+	Description    string         `json:"description,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/modelarts/v1/notebook/requests.go:66]: use pointer or string?

File: openstack/modelarts/v1/notebook/requests.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	url := mountDetailURL(c, id, storageId)
+	var rst MountStorage
+	_, err := c.DeleteWithResponse(url, &rst, &golangsdk.RequestOpts{
+		MoreHeaders: RequestOpts.MoreHeaders,
+	})
+
+	return &rst, err
+}
+
+func ListMount(c *golangsdk.ServiceClient, id string) (*MountStorageListResp, error) {
```
Comment (ShiChangkuo) [inline openstack/modelarts/v1/notebook/requests.go:267]: s/ListMount/ListMounts

File: pagination/offset.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		return "", err
+	}
+
+	offset = strconv.Itoa(int(offsetVal + sizeVal))
+	q.Set(offsetKey, offset)
+	currentURL.RawQuery = q.Encode()
+	return currentURL.String(), nil
+}
+
+// GetBody returns the page's body. This method is needed to satisfy the
```
Comment (ShiChangkuo) [inline pagination/offset.go:52]: IsEmpty method is missing?

Comment (hmbig2) [reply pagination/offset.go:52]: as response struct is different , implement the interface where it is used.

**Pull request merged**